### ISSUE TITLE
Library compile error on Data.swift

### ISSUE THREAD

**Issue by macellari-mac** [nordicsemi/IOS-DFU-Library#295]
I'm trying to test the library but I'm facing a compile error on file Data.swift

Line 36: 
$0.baseAddress!.bindMemory(to: R.self, capacity: 1).pointee
Error: /Users/luciomacellari/Projects/dfuDemo/Pods/iOSDFULibrary/iOSDFULibrary/Classes/Utilities/Data.swift:36:13: Value of type 'UnsafePointer<_>' has no member 'baseAddress'

Line 50:
array.append(contentsOf: $0)
Error:
/Users/luciomacellari/Projects/dfuDemo/Pods/iOSDFULibrary/iOSDFULibrary/Classes/Utilities/Data.swift:50:25: Extraneous argument label 'contentsOf:' in call

I'm compiling with Xcode 10.1

**Comment by KeerthiChinivar** [comment]
For **First issue** replace 

```swift
return subdata(in: offset ..< offset + length).withUnsafeBytes {
    $0.baseAddress!.bindMemory(to: R.self, capacity: 1).pointee
 }
```

with 

```swift
 #if swift(>=5.0)
    return subdata(in: offset ..< offset + length).withUnsafeBytes { $0.load(as: T.self) }
 #else
    return subdata(in: offset ..< offset + length).withUnsafeBytes { $0.pointee }
 #endif
```

For **Second issue** solution is:

```swift
var hexString: String {
    var array: [UInt8] = []
    self.withUnsafeBytes {
        array.append(contentsOf: getByteArray(pointer: $0))
    }
        
    return array.reduce("") { (result, byte) -> String in
        result + String(format: "%02x", byte)
    }
}
    
private func getByteArray(pointer: UnsafePointer<UInt8>) -> [UInt8] {
    let buffer = UnsafeBufferPointer<UInt8>(start: pointer, count: count)
        
    return [UInt8](buffer)
}
```

**Comment by philips77** [comment]
I will fix it in 4.4.2. @KeerthiChinivar thanks for your reply!
